### PR TITLE
Issue #2 create docs

### LINKED ISSUE #2: Create Documentation
To support users outside of Drashland, add documentation on where and how to create middlewares. For example, explain:

* File structure for new middlewares
* Usage of 3rd party modules
* Importing and exporting

This would also serve as self documentation for people within Drashland.

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #2 

Used most of the content from deno-drash but modified to fit this repo.

Added:

* DEV.md
* PR and Issue templates

Updated: 
* README

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    git checkout -b <topic-branch-name>
+    ```
+4. Create the directories and files to hold your middleware. Your middleware should be exported from it's own `mod.ts`:
+    ```bash
+    mkdir tests/<middleware>
+    mkdir src/<middleware>
+    touch src/<middleware>/mod.ts
+    touch src/<middleware>/related_files.ts
+    ```   
+    You should write extensive tests for your middleware
```
Comment (crookse) [inline .github/CONTRIBUTING.md:40]: Add in "A README.md file is REQUIRED"

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+3. Create a new topic branch (off the main project development branch) to contain your feature, change, or fix:
+    ```bash
+    git checkout -b <topic-branch-name>
+    ```
+4. Create the directories and files to hold your middleware. Your middleware should be exported from it's own `mod.ts`:
+    ```bash
+    mkdir tests/<middleware>
+    mkdir src/<middleware>
+    touch src/<middleware>/mod.ts
+    touch src/<middleware>/related_files.ts
```
Comment (crookse) [inline .github/CONTRIBUTING.md:38]: touch src/<middleware>/README.md

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Fixes {issue number}
```
Comment (crookse) [inline .github/pull_request_template.md:1]: Does "Fixes" close the related issue?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (crookse) [inline .gitignore:1]: .DS_Store as well

File: .github/pull_request_template.md (hunk 3 above)
Comment (ebebbington) [reply .github/pull_request_template.md:1]: Yeah if typed like `Fixes #5`, is there a better way to reference an issue? Might also change it to “Fixes|Implements”

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (ebebbington) [reply .github/CONTRIBUTING.md]: Added. See the closest commit below

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (ebebbington) [reply .github/CONTRIBUTING.md]: Added. see the closest commit below. I also removed the `src` bit after seeing your comments on the #1 PR

File: .gitignore (hunk 4 above)
Comment (ebebbington) [reply .gitignore:1]: Added. See the closest commit below

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ```bash
+    git checkout -b <topic-branch-name>
+    ```
+4. Create the directories and files to hold your middleware. Your middleware should be exported from it's own `mod.ts`:
+    ```bash
+    mkdir <middleware>
+    mkdir <middleware>/tests
+    touch <middleware>/mod.ts
+    touch <middleware>/related_files.ts
+    touch README.md
```
Comment (crookse) [inline .github/CONTRIBUTING.md:39]: <middleware>/README.md

**Pull request merged**